### DISCUSSION TITLE
pythonPath ignored in pyproject.toml (with pypy)

### DISCUSSION THREAD

**Discussion by jalelegenda** [microsoft/pyright#7063] [Q&A] [answered]
I have a python project that uses pypy. I also utilize a `pyproject.toml`. As you may be aware pypy uses a different name for the `site-packages` parent directory (it's pypy<version>), so setting a `venv` with a `venvPath` is failing for me.
So I tried setting `pythonPath = ./.venv/bin/python` and even to the absolute path. It will not work. Activating the environment prior to entering my editor (I use neovim) will allow pyright to resolve the paths correctly.
The pyright version is 1.1.347.
Does anyone know what I am doing wrong?
My `pyproject.toml`:

```
[tool.pyright]
include = ["src"]
executionEnvironments = [
    { root = "src" }
]
pythonPath = "./.venv/bin/python"
```

project structure:
```
├── .ropeproject
├── .venv
│   ├── bin
│   │   ├── Activate.ps1
│   │   ├── activate
│   │   ├── activate.csh
│   │   ├── activate.fish
│   │   ├── falcon-bench
│   │   ├── falcon-inspect-app
│   │   ├── falcon-print-routes
│   │   ├── pip
│   │   ├── pip3
│   │   ├── pip3.9
│   │   ├── py.test
│   │   ├── pypy -> python
│   │   ├── pypy3 -> python
│   │   ├── pytest
│   │   ├── python -> /home/jakov/.pyenv/versions/pypy3.9-7.3.15/bin/python
│   │   ├── python3 -> python
│   │   ├── python3.9 -> python
│   │   └── ruff
│   ├── include
│   ├── lib
│   │   └── pypy3.9
│   │       └── site-packages
│   ├── lib64 -> lib
│   └── pyvenv.cfg
├── Dockerfile
├── Makefile
├── poetry.lock
├── pyproject.toml
├── src
│   └── domain
│       ├── __init__.py
│       ├── entities.py
│       ├── events.py
│       ├── services.py
│       └── values.py
└── tests
    ├── conftest.py
    └── domain
        └── test_entities.py
```

**Comment by erictraut** [comment, marked as answer]
`pythonPath` is not a supported [configuration option](https://microsoft.github.io/pyright/#/configuration?id=main-configuration-options). 

There is a [language server setting](https://microsoft.github.io/pyright/#/settings) called `python.pythonPath`. I think that's what you're looking for.

**Comment by jalelegenda** [comment]
I feel like there's a strong case for making this a config option. It seems like a more convenient way than venv and venvPath for import resolution that would make a multi-root workspace management easier.
Of course I say this without knowing the exact technical challenges.

**Reply by erictraut** [reply]
The config file (whether it's `pyrightconfig.json` or `pyproject.toml`) is intended to be checked in and shared with a project's collaborators. The location of the python environment typically varies based on the developer's machine, so it probably doesn't make sense to support `pythonPath` in the config file. It makes more sense for each developer to set this in his or her LSP settings.

**Reply by jalelegenda** [reply]
How are `venv` and `venvPath` different?
